VideoWeb/VideoWeb/Controllers/ConsultationsController.cs

```diff
@@ -271,37 +200,53 @@ public async Task<IActionResult> StartConsultationAsync(StartPrivateConsultation
             }
         }
 
-        [HttpPost("end")]
-        [SwaggerOperation(OperationId = "LeaveConsultation")]
-        [ProducesResponseType((int) HttpStatusCode.OK)]
-        [ProducesResponseType((int) HttpStatusCode.NotFound)]
-        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
-        public async Task<IActionResult> LeaveConsultationAsync(LeavePrivateConsultationRequest request)
+        [HttpPost("lock")]
+        [SwaggerOperation(OperationId = "LockConsultationRoomRequest")]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> LockConsultationRoomRequestAsync(LockConsultationRoomRequest request)
         {
             try
             {
                 var conference = await GetConference(request.ConferenceId);
 
-                var participant = conference.Participants?.SingleOrDefault(x => x.Id == request.ParticipantId);
-                if (participant == null)
-                {
-                    _logger.LogWarning($"The participant with Id: {request.ParticipantId} is not found");
-                    return NotFound();
-                }
+                var lockRequestMapper = _mapperFactory.Get<LockConsultationRoomRequest, LockRoomRequest>();
+                var mappedRequest = lockRequestMapper.Map(request);
+                await _videoApiClient.LockRoomAsync(mappedRequest);
 
-                var leaveConsultationRequestMapper = _mapperFactory.Get<LeavePrivateConsultationRequest, LeaveConsultationRequest>();
-                var mappedRequest = leaveConsultationRequestMapper.Map(request);
-                await _videoApiClient.LeaveConsultationAsync(mappedRequest);
+                await NotifyRoomUpdateAsync(conference, new Room { Label = request.RoomLabel, Locked = request.Lock, ConferenceId = conference.Id });
 
-                return Ok();
+                return NoContent();
             }
             catch (VideoApiException e)
             {
-                _logger.LogError(e, $"End consultation error ConferenceId: {request.ConferenceId} and participant Id: {request.ParticipantId}, ErrorCode: {e.StatusCode}");
-                return StatusCode(e.StatusCode);
+                _logger.LogError(e, $"Could not update the lock state of the consultation room");
```

Comment:
No need for $ if there are no variables.
Also when using the logger avoid using the interpolated string. AppInsights requires the message template be a compile-time constant so that it can correctly pull out the values for `customDimensions` in the trace view

Reply:
fixed all in file